cmd/dependabot/internal/cmd/update.go

```diff
@@ -332,19 +332,20 @@ func processInput(input *model.Input, flags *UpdateFlags) {
 	if hasLocalAzureToken && !isGitSourceInCreds && azureRepo != nil {
 		log.Println("Inserting $LOCAL_AZURE_ACCESS_TOKEN into credentials")
 		log.Printf("Inserting artifacts credentials for %s organization.", azureRepo.Org)
+
+		// add both `dev.azure.com` and `org.visualstudio.com` credentials
 		input.Credentials = append(input.Credentials, model.Credential{
 			"type":     "git_source",
 			"host":     "dev.azure.com",
 			"username": "x-access-token",
 			"password": "$LOCAL_AZURE_ACCESS_TOKEN",
 		})
-		if len(input.Job.CredentialsMetadata) > 0 {
-			// Add the metadata since the next section will be skipped.
```

Comment:
This comment is incorrect; the next section is _not_ skipped and the `CredentialsMetadata` is correctly updated.  The test has been modified to verify this.